### DISCUSSION TITLE
How can I set a client shader?

### DISCUSSION THREAD

**Discussion by ryleu** [Minestom/Minestom#1098] [Help] [answered]
I want to create my own shader that makes the game look greyscale. How can I tell the client to use the shader whenever I want? Is that even possible? Thanks in advance.

**Comment by cody-quinn** [comment, marked as answer]
This is done with resource packs. You can send a player a server resource pack that is forced. 

https://minecraft.fandom.com/wiki/Shaders

**Reply by ryleu** [reply]
I know how to get the shader there, but how can I make the client actually use it? I don't want them to use it all the time.

**Reply by cody-quinn** [reply]
What do you mean by making the client actually use it? So long as the shader is in the resource pack on the top of your pack list it should be applied. 

Keep in mind some rendering mods interfere with this.

**Reply by ryleu** [reply]
I only want the client to use it when I want it to use it, otherwise it should not use the shader.

Basically, I want it enabled in some instances, but not others. The different instances have overlapping dimensions, so I can't use that.

**Reply by cody-quinn** [reply]
Ahhhhhh I got it. Sadly this isn't really that easy, minecraft doesn't just send that information. There are some shader Discord servers and resources that might help.

https://discord.gg/AHFep53RD4
https://github.com/McTsts/Minecraft-Shaders-Wiki
https://github.com/McTsts/mc-core-shaders

**Reply by ryleu** [reply]
Gotcha, that's gonna be interesting to figure out. Thanks!

**Reply by ryleu** [reply]
It seems like I can use the fog color to work as a switch for the shader.
